figo/utils.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+function parse_api_endpoint($api_endpoint)   {
+    $api_endpoint = rtrim($api_endpoint, "/");
+    if (substr($api_endpoint, 0, 8) != "https://")    {
+        $api_endpoint = "https://" . $api_endpoint;
+    }
+    $api_url = parse_url($api_endpoint);
+    if (array_key_exists("path", $api_url) == false)    {
+        $api_url["path"] = "";
+    }
+    return $api_url;
+}
```

Comment:
I think you need to close the `php` tag. I'm not sure how picky php is...

Reply:
according to this and my ide it seems to be a good practice to omit it: https://stackoverflow.com/questions/3219383/why-do-some-scripts-omit-the-closing-php-tag